modules/azure/loadbalancer.go

```diff
@@ -0,0 +1,201 @@
+package azure
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-09-01/network"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/stretchr/testify/require"
+)
+
+// IPType enumerator for IP Types, Public or Private.
+type IPType int
+
+// IPType values listed using iota
+const (
+	PublicIP IPType = iota
+	PrivateIP
+)
+
+// String values for IPType
+func (ipType IPType) String() string {
+	return [...]string{"PublicIP", "PrivateIP"}[ipType]
+}
+
+// LoadBalancerExists indicates whether the specified Load Balancer exists.
+// This function would fail the test if there is an error.
+func LoadBalancerExists(t testing.TestingT, loadBalancerName string, resourceGroupName string, subscriptionID string) bool {
+	exists, err := LoadBalancerExistsE(loadBalancerName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// LoadBalancerExistsE indicates whether the specified Load Balancer exists.
+func LoadBalancerExistsE(loadBalancerName string, resourceGroupName string, subscriptionID string) (bool, error) {
+	_, err := GetLoadBalancerE(loadBalancerName, resourceGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// GetLoadBalancerConfigNames gets a list of the Frontend Configuration Names for the Load Balancer.
+// This function would fail the test if there is an error.
+func GetLoadBalancerConfigNames(t testing.TestingT, loadBalancerName string, resourceGroupName string, subscriptionID string) []string {
+	configName, err := GetLoadBalancerConfigNamesE(loadBalancerName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return configName
+}
+
+// GetLoadBalancerConfigNamesE gets a list of the Frontend Configuration Names for the Load Balancer.
+func GetLoadBalancerConfigNamesE(loadBalancerName string, resourceGroupName string, subscriptionID string) ([]string, error) {
+	var configNames []string
+
+	lb, err := GetLoadBalancerE(loadBalancerName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return configNames, err
+	}
+
+	lbProps := lb.LoadBalancerPropertiesFormat
+	feConfigs := *lbProps.FrontendIPConfigurations
+	if len(feConfigs) == 0 {
+		// No Frontend Configurations present
+		return configNames, NewNotFoundError("Frontend Config", "Any", loadBalancerName)
+	}
+
+	// Get the names of the Frontend IP Configurations present
+	for _, config := range feConfigs {
```

Comment:
once you have checked length, instead of append (which will trigger one or multiple reallocs here, you can simply init the array or slice to the known length.

Reply:
Updated